compiler/src/dotty/tools/dotc/ast/tpd.scala

```diff
@@ -1018,17 +1021,26 @@ object tpd extends Trees.Instance[Type] with TypedTreeInfo {
    */
   class MapToUnderlying extends TreeMap {
     override def transform(tree: Tree)(implicit ctx: Context): Tree = tree match {
-      case tree: Ident if tree.symbol.exists && !tree.symbol.owner.isClass && !tree.symbol.isAnonymousFunction && skipLocal(tree.symbol) =>
+      case tree: Ident if isBinding(tree.symbol) && skipLocal(tree.symbol) =>
         tree.symbol.defTree match {
-          case defTree: ValOrDefDef if !defTree.rhs.isEmpty => transform(defTree.rhs)
+          case defTree: ValOrDefDef =>
+            val rhs = defTree.rhs
+            assert(!rhs.isEmpty)
+            transform(rhs)
           case _ => tree
         }
       case Inlined(_, _, arg) => transform(arg)
       case Block(Nil, arg) => transform(arg)
       case NamedArg(_, arg) => transform(arg)
       case tree => super.transform(tree)
     }
-    def skipLocal(sym: Symbol): Boolean = true
+
+    /** Should get the rhs of this binding */
+    protected def skipLocal(sym: Symbol): Boolean = true
+
+    /** Is this a symbol that of a local val or parameterless def for which we could get the rhs */
+    private def isBinding(sym: Symbol)(implicit ctx: Context): Boolean =
+      sym.exists && !sym.is(Param) && !sym.owner.isClass && !sym.isAnonymousFunction
```

Comment:
It seems the implementation does not test `parameterless` mentioned in the doc?

Reply:
I added the code to check it and a regression test